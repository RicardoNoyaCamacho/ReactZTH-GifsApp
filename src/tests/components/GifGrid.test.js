import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en componente <GifGrid />", () => {
  const category = "HolaMundo";
  test("Debe crear el componente GifGrid", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      
    ];

    useFetchGifs.mockReturnValue({
      data: [],
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
