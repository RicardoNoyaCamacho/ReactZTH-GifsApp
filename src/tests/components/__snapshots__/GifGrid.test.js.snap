// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid /> Debe crear el componente GifGrid 1`] = `
<Fragment>
  <h3>
    HolaMundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="cardGrid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Pruebas en componente <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    HolaMundo
  </h3>
  <div
    className="cardGrid animate__animated animate__fadeIn"
  />
</Fragment>
`;
